### PR TITLE
[YUNIKORN-571]Add a label to the placeholder pods the selector

### PR DESCRIPTION
**Pull request by HuangTing-Yao**
Rename `AnnotationPlaceholderFlag` to `PlaceholderFlag`, and add same label when placeholder created.

### CODE REVIEW

File: pkg/common/constants/constants.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ const AppManagerHandlerName = "yunikorn-app"
 const PlaceholderContainerImage = "k8s.gcr.io/pause"
 const PlaceholderContainerName = "pause"
 const PlaceholderPodRestartPolicy = "Never"
-const AnnotationPlaceholderFlag = "yunikorn.apache.org/placeholder"
+const PlaceholderFlag = "yunikorn.apache.org/placeholder"
```
Comment (wilfred-s) [inline pkg/common/constants/constants.go:58]: For the `ApplicationID` we use an unqualified `applicationId` in labels and a qualified version `yunikorn.apache.org/app-id` in the annotation. Shouldn't we follow the same rule here? I can live with either solution, I just think we need to set a standard for what we do.

File: pkg/common/utils/gang_utils.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func GetPlaceholderResourceRequest(resources map[string]resource.Quantity) v1.Re
 }
 
 func GetPlaceholderFlagFromPodSpec(pod *v1.Pod) bool {
-	if value, ok := pod.Annotations[constants.AnnotationPlaceholderFlag]; ok {
+	if value, ok := pod.Annotations[constants.PlaceholderFlag]; ok {
```
Comment (wilfred-s) [inline pkg/common/utils/gang_utils.go:78]: Do we add the label just as a convenience or is there more behind this?
The appID retrieval uses a logic to look at annotations and then labels (normal followed by spark).

**Review by wilfred-s — CHANGES_REQUESTED** [review]
Two points on consistency more than on the code itself.
We need to be clear and consistent in what we do.

File: pkg/common/constants/constants.go (hunk 1 above)
Comment (HuangTing-Yao) [reply pkg/common/constants/constants.go]: Sure, follow the some rule will more clear to read.

File: pkg/common/utils/gang_utils.go (hunk 2 above)
Comment (HuangTing-Yao) [reply pkg/common/utils/gang_utils.go]: I think we can follow the same logic in `GetApplicationIDFromPod()`, add the label part in `GetPlaceholderFlagFromPodSpec()`

**Review by wilfred-s — APPROVED** [review]
LGTM, please feel free to commit

**Pull request merged**